mm2src/coins_activation/src/tendermint_with_assets_activation.rs

```diff
@@ -217,6 +229,42 @@ impl From<TendermintInitError> for EnablePlatformCoinWithTokensError {
     }
 }
 
+async fn activate_with_walletconnect(
+    ctx: &MmArc,
+    chain_id: &str,
+    ticker: &str,
+    wallet_type: &mut TendermintWalletConnectionType,
+) -> MmResult<TendermintActivationPolicy, TendermintInitError> {
+    let account = {
+        let wc = WalletConnectCtx::from_ctx(ctx).expect("WalletConnectCtx should be initialized by now!");
+
+        if wc.is_ledger_connection().await {
+            *wallet_type = TendermintWalletConnectionType::WcLedger;
+        } else {
+            *wallet_type = TendermintWalletConnectionType::Wc;
+        };
```

Comment:
nit: IMO passing mutable references and modifying them like this is an anti-pattern. I would prefer returning this as part of the function's result and handling it on the caller side.

Reply:
thanks!, done